### PR TITLE
Create controller api; Rework files to be closer to boost guidelines.

### PR DESCRIPTION
**Pull request by mvpete**

### CODE REVIEW

File: include/boost/winasio/http/basic_http_queue.hpp [hunk 1]
```diff
@@ -14,15 +18,15 @@ namespace net = boost::asio;
 namespace winnet = net::windows;
 
 template <typename Executor = net::any_io_executor>
-class basic_http_handle : public winnet::basic_overlapped_handle<Executor> {
+class basic_http_queue : public winnet::basic_overlapped_handle<Executor> {
```
Comment (youyuanwu) [inline include/boost/winasio/http/basic_http_queue.hpp]: This is returned from:
HttpCreateHttpHandle
or HttpCreateRequestQueue
and is handle wrapper.

Maybe it is still good to name it basic_http_handle or basic_http_queue_handle?
The parent class is basic_overlapped_handle.

File: include/boost/winasio/http/basic_http_queue.hpp [hunk 2]
```diff
@@ -227,7 +231,15 @@ class basic_http_handle : public winnet::basic_overlapped_handle<Executor> {
     ec = boost::system::error_code(result,
                                    boost::asio::error::get_system_category());
   }
+
+  void shutdown(boost::system::error_code &ec) {
```
Comment (youyuanwu) [inline include/boost/winasio/http/basic_http_queue.hpp]: You need to rebase.

File: include/boost/winasio/http/http.hpp [hunk 3]
```diff
@@ -28,46 +32,13 @@ inline HANDLE open_raw_http_queue() {
...
-    BOOST_ASSERT(retCode == NO_ERROR);
-  }
-  ~http_initializer() {
-    DWORD retCode =
-        HttpTerminate(HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE_CONFIG, NULL);
-    BOOST_ASSERT(retCode == NO_ERROR);
-  }
-};
-
+using queue = basic_http_queue<net::io_context::executor_type>;
```
Comment (youyuanwu) [inline include/boost/winasio/http/http.hpp]: The default executor type should be any_executor.

File: include/boost/winasio/http/http_initializer.hpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#ifndef BOOST_WINASIO_HTTP_INITIALIZER_HPP
+#define BOOST_WINASIO_HTTP_INITIALIZER_HPP
+
```
Comment (youyuanwu) [inline include/boost/winasio/http/http_initializer.hpp]: Usually #pragma once is also added here inside the #define

**Review by youyuanwu — CHANGES_REQUESTED** [review]
Please rebase to main branch.

File: include/boost/winasio/http/http_initializer.hpp (hunk 4 above)
Comment (mvpete) [reply include/boost/winasio/http/http_initializer.hpp]: Ah.
```
#if defined(_MSC_VER) && (_MSC_VER >= 1200)
# pragma once
#endif // defined(_MSC_VER) && (_MSC_VER >= 1200)
```

**Pull request closed without merging**